### PR TITLE
Add "Implant" to list of uplink locations

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: You may now pick Implant as a preferred uplink location, meaning
that if you become a traitor, you will start with an uplink implant. The
cost of the implant (4 TC) is deducted from your total starting TC,
because the Syndicate doesn't give freebies.
fix: Admins removing an uplink from a person with an uplink implant will
now destroy the implant, rather than leaving an implant that does
nothing.
/:cl:

If you are the sort of person who gets their PDA, pen and radio stolen
really easily, then maybe you'd be interested in just spawning with an
implant. Be warned, all of the 17-20 TC items will be out of reach
because of the implant cost.

Could be considered slightly more powerful than just ordering the
implant at shift start, given you don't have to get rid of an implanter
and suspicious box anymore.

When testing with the admin grant uplink buttons, I discovered that the
"Take" uplink option was only half working when taking the uplink from a
person with only an implant uplink. It would take the component, but not
the implant, leaving the person with a do-nothing action button. This
has been fixed.

### CODE REVIEW

**Review by Cyberboss — DISMISSED** [review]

File: code/_globalvars/lists/flavor_misc.dm [hunk 1]
```diff
@@ -160,7 +160,8 @@ GLOBAL_LIST_INIT(jumpsuitlist, list(PREF_SUIT, PREF_SKIRT))
 #define UPLINK_PDA		"PDA"
 #define UPLINK_RADIO	"Radio"
 #define UPLINK_PEN		"Pen" //like a real spy!
-GLOBAL_LIST_INIT(uplink_spawn_loc_list, list(UPLINK_PDA, UPLINK_RADIO, UPLINK_PEN))
+#define UPLINK_IMPLANT  "Implant"  // expensive, but unstealable
```
Comment (bobbahbrown) [inline code/_globalvars/lists/flavor_misc.dm:163]: Please document this as you would document a variable with dmdoc to be compliant with our requirements for documentation. [More information can be found here.](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)

File: code/game/objects/items/implants/implantuplink.dm [hunk 2]
```diff
@@ -10,6 +10,16 @@
 /obj/item/implant/uplink/Initialize(mapload, _owner)
 	. = ..()
 	AddComponent(/datum/component/uplink, _owner, TRUE, FALSE, null, starting_tc)
+	RegisterSignal(src, COMSIG_COMPONENT_REMOVING, .proc/_component_removal)
+
+/obj/item/implant/uplink/proc/_component_removal(datum/source, datum/component/component)
+	/**
```
Comment (bobbahbrown) [inline code/game/objects/items/implants/implantuplink.dm:16]: This documentation is incorrectly executed, the documentation should be on the line above the proc (line 14 in this current version)

[More information on our documentation requirements can be found here](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)

**Comment by ATH1909** [comment]
Would it be possible to make this account for the TC savings that'd occur if uplink implants happen to be discounted?

**Comment by Shadowflame909** [comment]
>Would it be possible to make this account for the TC savings that'd occur if uplink implants happen to be discounted that shift?

That's a lot of work for a rare situation

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Docs

**Comment by ExcessiveUseOfCobblestone** [comment]
it is part of the tradeoff for having something you can presumably take with you to jail.

File: code/game/objects/items/implants/implantuplink.dm [hunk 3]
```diff
@@ -21,3 +31,6 @@
 
 /obj/item/implant/uplink/precharged
 	starting_tc = 10
+
+/obj/item/implant/uplink/starting
+	starting_tc = 16  /// default TC count 20, minus cost of implant (4)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/implants/implantuplink.dm:36]: there is a potential issue here where the uplink implant cost gets changed and they forget to adjust this.

Comment (Cyberboss) [reply code/game/objects/items/implants/implantuplink.dm:36]: Yeah, should be defines somewhere for default starting TC and the cost of the implant. If not, create them please

**Comment by coiax** [comment]
I'll come back this, I promise.

File: code/__DEFINES/antagonists.dm [hunk 4]
```diff
@@ -87,3 +87,12 @@ GLOBAL_LIST_INIT(heretic_start_knowledge,list(/datum/eldritch_knowledge/spell/ba
 
 /// How much does it cost to reroll strains?
 #define BLOB_REROLL_COST 40
+
+/// How many telecrystals a normal traitor starts with
+#define TELECRYSTALS_DEFAULT 20
```
Comment (PKPenguin321) [inline code/__DEFINES/antagonists.dm:92]: Are we sure there isn't already a way to pull this value from the code somewhere without a new define? Seems very odd to make a new define like this but not actually apply it in the code where uplinks are given 20 TCs

**Comment by Cyberboss** [comment]
Haven't I seen this PR befo... yep

File: code/__DEFINES/antagonists.dm (hunk 4 above)
Comment (coiax) [reply code/__DEFINES/antagonists.dm:92]: You are totally correct, I forgot to use it in the actual place where the 20 TC is defined.

**Pull request merged**